lib/client/src/client/client.ts

```diff
@@ -138,15 +197,21 @@ export interface Client<R extends Range> {
      * like RxJS), but it means that the client needs to manually poll for updated annotations if
      * freshness is important.
      */
-    annotations(params: AnnotationsParams): Promise<Annotation<R>[]>
+    annotations(
+        params: AnnotationsParams,
+        providerUri?: string
+    ): Promise<EachWithProviderUri<Annotation<R>[]>>
 
     /**
      * Observe OpenCtx annotations from the configured providers for the given resource.
      *
      * The returned observable streams annotations as they are received from the providers and
      * continues passing along any updates until unsubscribed.
      */
-    annotationsChanges(params: AnnotationsParams): Observable<Annotation<R>[]>
+    annotationsChanges(
+        params: AnnotationsParams,
+        providerUri?: string
```

Comment:
What is the purpose of `providerUri`? Asking for myself now here and also you should add it in the docstring.

Why do you need to change the annotations API given that this work is about items?

Reply:
The usage of `provider` is to get items/annotations for a specific provider. `item.kind` is expected to be unique only at the provider level, not globally. `providerUri` is the identifier for the provider. It value is based on the provider path the user sets in the OpenCtx config. 

We can skip making any changed to annotations for now.

Reply:
Yeah lets follow-up. I believe its added here so you can route a request to a specific provider which maybe doesn't make sense for annotations, but is useful from a consistency in API point of view. So for example when passing in a mention we make sure to send it to the correct provider.